common/src/main/java/com/wynntils/features/user/tooltips/ItemCompareFeature.java

```diff
@@ -0,0 +1,167 @@
+
+    @RegisterKeyBind
+    private final KeyHolder compareSelectKeybind =
+            new KeyHolder("Select for comparing", GLFW.GLFW_KEY_C, "Wynntils", true, () -> {});
+
+    private static GearItemStack COMPARED_ITEM = null;
+    private static boolean COMPARE_MODE_ON = false;
+
+    @SubscribeEvent
+    public void onRenderSlot(SlotRenderEvent.Pre event) {
+        Slot slot = event.getSlot();
+        if (slot.getItem() == COMPARED_ITEM) {
+            RenderUtils.drawArc(CommonColors.LIGHT_BLUE, slot.x, slot.y, 200, 1, 6, 8);
+        }
+    }
+
+    @SubscribeEvent
+    public void onItemTooltipRenderEvent(ItemTooltipRenderEvent.Post event) {
+        if (!(McUtils.mc().screen instanceof AbstractContainerScreen<?> abstractContainerScreen)) return;
+        if (!COMPARE_MODE_ON) return;
+
+        if (abstractContainerScreen.hoveredSlot == null) {
+            return;
+        }
+
+        ItemStack hoveredItemStack = abstractContainerScreen.hoveredSlot.getItem();
+
+        if (event.getItemStack() != hoveredItemStack) {
+            return;
+        }
+
+        if (!(hoveredItemStack instanceof GearItemStack hoveredGearItemStack)) {
+            return;
+        }
+
+        // No compared item selected, try compare to equipped armor
+        if (COMPARED_ITEM == null) {
+            List<ItemStack> armorSlots = McUtils.player().getInventory().armor;
+
+            Optional<GearItemStack> matchingArmorItemStack = armorSlots.stream()
+                    .filter(itemStack -> itemStack instanceof GearItemStack gItemStack
+                            && gItemStack.getItemProfile().getItemInfo().getType()
+                                    == hoveredGearItemStack
+                                            .getItemProfile()
+                                            .getItemInfo()
+                                            .getType())
+                    .map(itemStack -> (GearItemStack) itemStack)
+                    .findFirst();
+
+            matchingArmorItemStack.ifPresent(gearItemStack -> handleCompareTo(
+                    new PoseStack(),
+                    event.getMouseX(),
+                    event.getMouseY(),
+                    abstractContainerScreen,
+                    hoveredGearItemStack,
+                    gearItemStack));
+        } else {
+            handleCompareTo(
+                    new PoseStack(),
+                    event.getMouseX(),
+                    event.getMouseY(),
+                    abstractContainerScreen,
+                    hoveredGearItemStack,
+                    COMPARED_ITEM);
+        }
+    }
+
+    private static void handleCompareTo(
+            PoseStack poseStack,
+            int mouseX,
+            int mouseY,
+            AbstractContainerScreen<?> inventoryScreen,
+            GearItemStack hoveredGearItemStack,
+            GearItemStack matchingArmorItemStack) {
+        if (hoveredGearItemStack == matchingArmorItemStack) return;
```

Comment:
This can't happen, can it? I don't think you need to protect for that. And if it *can* happen, there's something wrong with the logic in onItemTooltipRenderEvent, and we should abort much earlier.